liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -60,120 +72,133 @@ protected void installApp(Artifact artifact) throws Exception {
         } else {
             copyFile.setTodir(destDir);
         }
-        
-        // validate application configuration if appsDirectory="dropins" or inject webApplication
+
+        // validate application configuration if appsDirectory="dropins" or inject
+        // webApplication
         // to target server.xml if not found for appsDirectory="apps"
-        validateAppConfig(fileName, artifact.getArtifactId());
-        
+        validateAppConfig(fileName, artifact.getArtifactId(), thin);
+
         deleteApplication(new File(serverDirectory, "apps"), artifact.getFile());
         deleteApplication(new File(serverDirectory, "dropins"), artifact.getFile());
-        // application can be expanded if server.xml configure with <applicationManager autoExpand="true"/>
+        deleteApplication(new File(serverDirectory, "dropins/spring"), artifact.getFile());
+        // application can be expanded if server.xml configure with <applicationManager
+        // autoExpand="true"/>
         deleteApplication(new File(serverDirectory, "apps/expanded"), artifact.getFile());
         copyFile.execute();
     }
-    
+
+    private void copyLibIndexCache() throws IOException {
+        String libIndexCacheName = "lib.index.cache";
+        File sourceLibDir = new File(project.getBuild().getDirectory(), libIndexCacheName);
```

Comment:
Should we check if this file exists and throw an exception if not?

Reply:
Done.